src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs

```diff
@@ -171,6 +171,16 @@ public override SyncResponseHandlingResult HandleResponse(BodiesSyncBatch? batch
                     ? SyncResponseHandlingResult.NoProgress
                     : SyncResponseHandlingResult.OK;
             }
+            catch (Exception)
+            {
+                foreach (BlockInfo batchInfo in batch.Infos)
+                {
+                    if (batchInfo == null) break;
```

Comment:
Either mark properly nullable or use like `.Where(i => i is not null)` Also we are preferring `is not null`